agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -455,47 +444,58 @@ impl DeriveSchema for Stage {
                 state.result_set_schema = Schema::simplify(&state.result_set_schema);
                 return Ok(state.result_set_schema.to_owned());
             }
+            let mut result_doc = Schema::Document(Document::empty());
```

Comment:
still work to be done ofc, but the crux of the $project changes here is rather than collecting key-schema pairs, we just create an empty document and insert the full paths as we go. I suppose we could collect path-schema pairs, but its the same either way 🤷